assets/model_monitoring/components/generation_safety_quality/annotation_compute_histogram/spec.yaml

```diff
@@ -0,0 +1,124 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+type: spark
+
+name: annotation_compute_histogram_spark
+display_name: Annotation - Compute Histogram
+description: Compute annotation histogram given a deployment's model data input.
+version: 0.3.4
+is_deterministic: false
+inputs:
+  target_dataset: 
+    type: mltable
+    mode: direct
+  metric_names:
+    type: string
+    description: a comma-separated list of metric names to compute
+    default: groundedness
+  model_type:
+    type: string
+    optional: false
+    description: type of model used for annotation
+    enum:
+      - gpt-35-turbo
+      - gpt-4
+      - gpt-4-32k
+  model_deployment_name:
+    type: string
+    optional: false
+    description: model name, e.g., name of the deployment for Azure OpenAI
+  endpoint_type:
+    type: string
+    optional: true
+    enum:
+      - openai_api
```

Comment:
I believe this is kind of confusing now... the other options only exist for legacy reasons right now and don't actually work. Maybe we should remove them until they're actually supported? As in, only leave the default.

Reply:
azure_endpoint and azure_openai_api don't work ? Let me bring this up with Will

Reply:
Yeah if you check the validation logic... it throws errors.